File: Java/Medium/Accepted/string-compression_7.java
class Solution {
    public int compress(char[] chars) {
        
        if (chars.length < 2) return chars.length;
        int i = 0, j = 0, k = 0;
        while (j < chars.length) {
            while (j < chars.length && chars[i] == chars[j]) {
                j++;
            }
            chars[k++] = chars[i];
            if (j - i > 1) {
                for (char c : String.valueOf(j - i).toCharArray()) {
                    chars[k++] = c;
                }
            }
            i = j;
        }
        return k;
    }
}

File: C++/Medium/RuntimeError/string-compression_7.cpp
class Solution {
public:
    int compress(vector<char>& chars) {
        
        if (chars.size() <= 1) {
            return chars.size();
        }
        int slow = 0;
        int fast = 0;
        while (fast < chars.size()) {
            int count = 0;
            while (fast < chars.size() && chars[fast] == chars[slow]) {
                fast++;
                count++;
            }
            chars[slow] = chars[slow];
            slow++;
            if (count == 1) {
                continue;
            }
            string countStr = to_string(count);
            for (char c : countStr) {
                chars[slow] = c;
                slow++;
            }
        }
        return slow;
    }
};